File: scripts/test_solutions/1448_count_good_nodes.py
def goodNodes(root):
    count = [0]

    def dfs(node, max_val):
        if not node:
            return
        if node.val >= max_val:
            count[0] += 1
        max_val = max(max_val, node.val)
        dfs(node.left, max_val)
        dfs(node.right, max_val)

    dfs(root, float('-inf'))
    return count[0]

File: scripts/test_solutions/1448_count_good_nodes.cpp
class Solution {
public:
    int goodNodes(TreeNode* root) {
        return dfs(root, INT_MIN);
    }

private:
    int dfs(TreeNode* node, int maxVal) {
        if (!node) return 0;

        int count = 0;
        if (node->val >= maxVal) {
            count = 1;
        }

        maxVal = max(maxVal, node->val);
        count += dfs(node->left, maxVal);
        count += dfs(node->right, maxVal);

        return count;
    }
};
